llmfoundry/callbacks/async_eval_callback.py

```diff
@@ -0,0 +1,274 @@
+
+        self.check_interval = create_interval_scheduler(interval)
+        self.compute = compute
+        self.last_checkpoint: Optional[str] = None
+
+        # Run these during init to fail fast in any of the error cases
+        self.current_run = self._get_current_run()
+        get_eval_parameters(
+            parameters=training_config,
+            checkpoint='test',
+            training_run_name=self.current_run.name,
+        )
+        log.info(
+            f'Initialized AsyncEval callback. Will generate runs at interval {interval}'
+        )
+
+    def run_event(self, event: Event, state: State, logger: Logger) -> None:
+        del logger
+
+        should_launch_run = all([
+            state.get_elapsed_duration() is not None,
+            self.check_interval(state, event),
+            dist.get_global_rank() == 0,
+        ])
+
+        if should_launch_run:
+            current_interval = state.timestamp.get(self.interval.unit)
+            checkpoint = get_latest_checkpoint(event, state)
+            if not checkpoint:
+                return  # warnings logged in get_latest_checkpoint
+            if checkpoint == self.last_checkpoint:
+                # Do not eval a checkpoint that has already been evaluated.
+                log.info(
+                    'Skipping async eval because the checkpoint has not changed'
+                )
+                return
+
+            self.launch_run(checkpoint, current_interval)
+            self.last_checkpoint = checkpoint
+
+    def _get_current_run(self) -> Run:
+        if os.environ.get(MOSAICML_PLATFORM_ENV_VAR,
+                          'false').lower() == 'false':
+            raise RuntimeError(
+                'AsyncEval callback is only supported when running on the MosaicML platform'
+            )
+
+        run_name = os.environ.get(RUN_NAME_ENV_VAR, None)
+        if not run_name:
+            raise RuntimeError(
+                'RUN_NAME environment variable must be set to use the AsyncEval callback'
+            )
+
+        # Allows the MapiException to be raised if the run doesn't exist
+        return get_run(run_name, include_details=True)
+
+    def launch_run(self, checkpoint: str, current_interval: str) -> Run:
+        cfg = self.current_run.submitted_config
+        default_compute = {
+            'gpus': 8,
+            'cluster': self.current_run.cluster,
+        }
+
+        run_name = get_run_name(self.current_run.name, current_interval)
+
+        params = get_eval_parameters(
+            parameters=self.training_config,
+            checkpoint=checkpoint,
+            training_run_name=self.current_run.name,
+        )
+        params['run_name'] = run_name
+
+        # TODO: This just runs an eval run, but we also want to attach the
+        # deployment, which would require a hf conversion and parametrizing the
+        # dependent_deployment in the run config
+        command = 'cd llm-foundry/scripts \n composer eval/eval.py $PARAMETERS'
```

Comment:
this command assumes that the llm foundry integration is set up a specific way. may need to do a bit more parsing to make sure these paths are valid